### PR TITLE
Fix cache error when getting invalid files

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #9074

https://pulp.plan.io/issues/9074
Thanks @lubosmj for finding this bug.

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tests/unit/test_cache.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def test_03_basic_delete(self):
 
     def test_04_basic_expires(self):
         """Tests setting values with expiration times"""
+        self.skipTest("Timing is inconsistent in CI")
```
Comment (dralley) [inline pulpcore/tests/unit/test_cache.py:36]: Make a test issue about this, if there isn't one already

**Review by dralley — APPROVED** [review]

**Comment by dralley** [comment]
@gerrod3 Do you think you could add a small test for this?

File: pulpcore/tests/unit/test_cache.py (hunk 1 above)
Comment (dralley) [reply pulpcore/tests/unit/test_cache.py:36]: Just update it to mention that it's currently disabled

**Review by ipanova — APPROVED** [review]

File: pulpcore/tests/functional/api/using_plugin/test_content_cache.py [hunk 2]
```diff
@@ -134,9 +134,16 @@ def test_09_delete_repo_invalidates(self):
...
-        for i, file in enumerate(files):
+        for file in files:
             self.assertEqual((404, None), self.check_cache(file), file)
 
+    def test_10_no_error_when_accessing_invalid_file(self):
+        """Tests that accessing a file that doesn't exist on content app gives 404"""
+        files = ["invalid", "another/bad-one", "DNE/"]
+        url = PULP_CONTENT_BASE_URL
+        for file in files:
+            self.assertEqual((404, None), self.check_cache(file, url=url))
```
Comment (dralley) [inline pulpcore/tests/functional/api/using_plugin/test_content_cache.py:145]: It looks like we're not consistently including "file" as a third parameter (which allows a test runner to see which file the test failed on).  Could you add it here and in any other places?  I see some on lines 76, 94, 106, 107

**Review by dralley — APPROVED** [review]

**Pull request merged**